ophyd/areadetector/plugins.py

```diff
@@ -106,6 +112,15 @@ def array_pixels(self):
 
         return pixels
 
+    def read_configuration(self):
+        ret = super().read_configuration()
+
+        source_port = self.nd_array_port.get()
+        source_plugin = self.root.get_plugin_by_port(source_port)
+        ret[source_port] = source_plugin.read_configuration()
```

Comment:
The idea was to to directly encode the ordering in the nesting.  Get the configuration of the parent via `dd[dd['nd_array_port']]`.

I do not think we want to merge these dictionaries as I think eventually will have `understand_what_stats_plugin_was_doing()` functions on the analysis side and it would be nice to not have to allow those to accept (but ignore) extra keys.